src/xpk/core/capacity.py

```diff
@@ -284,27 +285,59 @@ def get_capacity_node_selectors_from_capacity_type(
   return node_selector, return_code
 
 
-def get_reservation_project_and_name(
-    reservation_name_or_path: str, cluster_project: str
-) -> tuple[str, str]:
-  """Get the reservation project and name.
+def parse_reservation(
+    reservation_path: str, cluster_project: str
+) -> Reservation:
+  """Parses the reservation details from the reservation path.
+      Also supports reservation blocks and sub-blocks.
+      Assumes cluster project if project is not contained in the path.
 
-  Args:
-    reservation_name_or_path: either reservation name or reservation path in format
-      projects/RESERVATION_PROJECT_ID/reservations/RESERVATION_NAME
-    cluster_project: the cluster project
+      Args:
+        reservation_path: path to the reservation, reservation block or sub-block in format:
+  `[projects/RESERVATION_PROJECT_ID/reservations/]RESERVATION_NAME[/reservationBlocks/BLOCK_NAME[/reservationSubBlocks/SUB_BLOCK_NAME]]`
+        cluster_project: the cluster project
 
-  Returns:
-    Tuple with reservation project and reservation name.
+      Returns:
+        Reservation instance containing reservation details.
   """
-  if '/' not in reservation_name_or_path:
-    return cluster_project, reservation_name_or_path
-  reservation_parts = reservation_name_or_path.split('/')
-  if (
-      len(reservation_parts) != 4
-      or reservation_parts[0] != 'projects'
-      or reservation_parts[2] != 'reservations'
-  ):
-    xpk_print('Unable to parse reservation: ', reservation_name_or_path)
+  reservation = _try_parse_reservation(reservation_path, cluster_project)
+  if reservation is None:
+    xpk_print('Unable to parse reservation: ', reservation_path)
     xpk_exit(1)
-  return reservation_parts[1], reservation_parts[3]
+  return reservation
+
+
+def _try_parse_reservation(
+    reservation_path: str, cluster_project: str
+) -> Reservation | None:
+  # assume trivial case, path contains just the reservation name
+  reservation = Reservation(
+      project=cluster_project,
+      name=reservation_path,
+      block_name=None,
+      sub_block_name=None,
+  )
+  parts = reservation_path.split('/')
+  if min(map(len, parts)) == 0:  # all parts must be non-empty
+    return None
+  if len(parts) == 1:
+    return reservation  # trivial case
+
+  if parts[0] == 'projects':
```

Comment:
Could we create a helper that can parse any oneplatformlink to a dict, it could be useful later, e.g. `foo/foo-1/bar/bar-1` would result in `{'foo': 'foo-1', 'bar': 'bar-1'}`. Although, it's a bit less strict, as it doesn't enforce the order.